### ISSUE TITLE
Bug: before() called before handles from waitOn are ready

### ISSUE THREAD

**Issue by gadicc** [iron-meteor/iron-router#336]
According to the docs:

> When your route is run, it will wait on any subscriptions you've provided in your waitOn function before running your **before hooks**, action method, and after hooks.

But this simple test shows otherwise (in 0.6.1):

``` js
  Router.map(function() {
    this.route('/', {
      template: 'hello',
      loadingTemplate: 'loading',
      waitOn: { ready: function() { return false; }}, // non reactive false function
      before: function() { alert('before'); }, // should never happen, but does
      data: function() { alert('data'); } // never happens, as expected
    })
  });
```

Not sure if this is a bug or the docs are wrong (since it looks like `before()` used with `.wait()` can fulfill a similar function -- but it would still be nice to have some subscriptions ready before executing code relevant for a before hook.

Thanks :)
